In [1]:
import pandas as pd
import pathlib

In [2]:
dataset_path = pathlib.Path.cwd()/'Datasets'
WHS_path = dataset_path/'World_Health_Statistics_archive'

## Cancer

In [3]:
cancer_data = pd.read_csv(WHS_path/'30-70cancerChdEtc.csv')

In [4]:
cancer_data

Location  Period                                          Indicator  \
0     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
1     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
2     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
3     Afghanistan    2015  Probability (%) of dying between age 30 and ex...   
4     Afghanistan    2015  Probability (%) of dying between age 30 and ex...   
...           ...     ...                                                ...   
2740     Zimbabwe    2005  Probability (%) of dying between age 30 and ex...   
2741     Zimbabwe    2005  Probability (%) of dying between age 30 and ex...   
2742     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   
2743     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   
2744     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   

            Dim1  First Tooltip  
0     Both sexes           29.8  
1           Male           31.8  
2         Female           27.7  
3     Both sexes           29.8  
4           Male           31.9  
...          ...            ...  
2740        Male           22.1  
2741      Female           22.9  
2742  Both sexes           21.6  
2743        Male           21.6  
2744      Female           21.7  

[2745 rows x 5 columns]

There is no nulls in this dataset

In [5]:
cancer_data.isnull().count()

Location         2745
Period           2745
Indicator        2745
Dim1             2745
First Tooltip    2745
dtype: int64

In [191]:
cancer_data[cancer_data['First Tooltip'] == 0.0].count()

Location         0
Period           0
Indicator        0
Dim1             0
First Tooltip    0
dtype: int64

The years in this dataset are 2000, 2005, 2010, 2015, and 2016. It appears to be every 5 years except in regards to 2016.

In [6]:
cancer_data['Period'].unique()

array([2016, 2015, 2010, 2005, 2000])

All counts are the same for each year, so any year is a great choice.

In [7]:
cancer_data[cancer_data['Period'] == 2016].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [8]:
cancer_data[cancer_data['Period'] == 2015].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [9]:
cancer_data[cancer_data['Period'] == 2010].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [10]:
cancer_data[cancer_data['Period'] == 2005].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [11]:
cancer_data[cancer_data['Period'] == 2000].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

All data is about dying between the age 30 and 70 from cardiovascular disease, cancer, diabetes, or a chronic respiratory disease.

In [12]:
cancer_data['Indicator'].unique()

array(['Probability (%) of dying between age 30 and exact age 70 from any of cardiovascular disease, cancer, diabetes, or chronic respiratory disease'],
      dtype=object)

In [13]:
#cancer_data['Dim1'].str.contains("Both Sexes").count()

In [14]:
#cancer_data.iloc[[0]]

In [15]:
cancer_both_sex = cancer_data[cancer_data['Dim1'] == 'Both sexes']
cancer_both_sex

Location  Period                                          Indicator  \
0     Afghanistan    2016  Probability (%) of dying between age 30 and ex...   
3     Afghanistan    2015  Probability (%) of dying between age 30 and ex...   
6     Afghanistan    2010  Probability (%) of dying between age 30 and ex...   
9     Afghanistan    2005  Probability (%) of dying between age 30 and ex...   
12    Afghanistan    2000  Probability (%) of dying between age 30 and ex...   
...           ...     ...                                                ...   
2730     Zimbabwe    2016  Probability (%) of dying between age 30 and ex...   
2733     Zimbabwe    2015  Probability (%) of dying between age 30 and ex...   
2736     Zimbabwe    2010  Probability (%) of dying between age 30 and ex...   
2739     Zimbabwe    2005  Probability (%) of dying between age 30 and ex...   
2742     Zimbabwe    2000  Probability (%) of dying between age 30 and ex...   

            Dim1  First Tooltip  
0     Both sexes           29.8  
3     Both sexes           29.8  
6     Both sexes           31.7  
9     Both sexes           34.1  
12    Both sexes           34.4  
...          ...            ...  
2730  Both sexes           19.3  
2733  Both sexes           19.4  
2736  Both sexes           21.1  
2739  Both sexes           22.5  
2742  Both sexes           21.6  

[915 rows x 5 columns]

## This dataframe will need to be exported.

In [16]:
cancer_both_sex_2016 = cancer_both_sex[cancer_both_sex['Period'] == 2016]
cancer_both_sex_2016

Location  Period  \
0                            Afghanistan    2016   
15                               Albania    2016   
30                               Algeria    2016   
45                                Angola    2016   
60                   Antigua and Barbuda    2016   
...                                  ...     ...   
2670  Venezuela (Bolivarian Republic of)    2016   
2685                            Viet Nam    2016   
2700                               Yemen    2016   
2715                              Zambia    2016   
2730                            Zimbabwe    2016   

                                              Indicator        Dim1  \
0     Probability (%) of dying between age 30 and ex...  Both sexes   
15    Probability (%) of dying between age 30 and ex...  Both sexes   
30    Probability (%) of dying between age 30 and ex...  Both sexes   
45    Probability (%) of dying between age 30 and ex...  Both sexes   
60    Probability (%) of dying between age 30 and ex...  Both sexes   
...                                                 ...         ...   
2670  Probability (%) of dying between age 30 and ex...  Both sexes   
2685  Probability (%) of dying between age 30 and ex...  Both sexes   
2700  Probability (%) of dying between age 30 and ex...  Both sexes   
2715  Probability (%) of dying between age 30 and ex...  Both sexes   
2730  Probability (%) of dying between age 30 and ex...  Both sexes   

      First Tooltip  
0              29.8  
15             17.0  
30             14.2  
45             16.5  
60             22.6  
...             ...  
2670           18.1  
2685           17.1  
2700           30.6  
2715           17.9  
2730           19.3  

[183 rows x 5 columns]

## This dataframe will need to be exported.

In [17]:
cancer_both_sex_2015 = cancer_both_sex[cancer_both_sex['Period'] == 2015]
cancer_both_sex_2015

Location  Period  \
3                            Afghanistan    2015   
18                               Albania    2015   
33                               Algeria    2015   
48                                Angola    2015   
63                   Antigua and Barbuda    2015   
...                                  ...     ...   
2673  Venezuela (Bolivarian Republic of)    2015   
2688                            Viet Nam    2015   
2703                               Yemen    2015   
2718                              Zambia    2015   
2733                            Zimbabwe    2015   

                                              Indicator        Dim1  \
3     Probability (%) of dying between age 30 and ex...  Both sexes   
18    Probability (%) of dying between age 30 and ex...  Both sexes   
33    Probability (%) of dying between age 30 and ex...  Both sexes   
48    Probability (%) of dying between age 30 and ex...  Both sexes   
63    Probability (%) of dying between age 30 and ex...  Both sexes   
...                                                 ...         ...   
2673  Probability (%) of dying between age 30 and ex...  Both sexes   
2688  Probability (%) of dying between age 30 and ex...  Both sexes   
2703  Probability (%) of dying between age 30 and ex...  Both sexes   
2718  Probability (%) of dying between age 30 and ex...  Both sexes   
2733  Probability (%) of dying between age 30 and ex...  Both sexes   

      First Tooltip  
3              29.8  
18             17.4  
33             14.4  
48             16.9  
63             22.6  
...             ...  
2673           18.2  
2688           17.3  
2703           30.7  
2718           18.0  
2733           19.4  

[183 rows x 5 columns]

## Adolescent Birth Rate

In [194]:
ad_br_data = pd.read_csv(WHS_path/'adolescentBirthRate.csv')
ad_br_data

Location  Period                                          Indicator  \
0     Afghanistan    2017  Adolescent birth rate (per 1000 women aged 15-...   
1     Afghanistan    2014  Adolescent birth rate (per 1000 women aged 15-...   
2     Afghanistan    2013  Adolescent birth rate (per 1000 women aged 15-...   
3     Afghanistan    2011  Adolescent birth rate (per 1000 women aged 15-...   
4     Afghanistan    2009  Adolescent birth rate (per 1000 women aged 15-...   
...           ...     ...                                                ...   
2184     Zimbabwe    2008  Adolescent birth rate (per 1000 women aged 15-...   
2185     Zimbabwe    2007  Adolescent birth rate (per 1000 women aged 15-...   
2186     Zimbabwe    2004  Adolescent birth rate (per 1000 women aged 15-...   
2187     Zimbabwe    2003  Adolescent birth rate (per 1000 women aged 15-...   
2188     Zimbabwe    2002  Adolescent birth rate (per 1000 women aged 15-...   

      First Tooltip  
0              62.0  
1              77.2  
2              87.0  
3             125.7  
4              80.0  
...             ...  
2184          112.0  
2185          102.6  
2186           98.7  
2187          101.0  
2188          102.6  

[2189 rows x 4 columns]

In [189]:
ad_br_data.isnull().count()

Location         2189
Period           2189
Indicator        2189
First Tooltip    2189
dtype: int64

In [190]:
ad_br_data[ad_br_data['First Tooltip'] == 0.0].count()

Location         1
Period           1
Indicator        1
First Tooltip    1
dtype: int64

In [196]:
ad_br_data[ad_br_data['First Tooltip'] == 0.0]

Location  Period                                          Indicator  \
1707  San Marino    2012  Adolescent birth rate (per 1000 women aged 15-...   

      First Tooltip  
1707            0.0

In [20]:
ad_br_data['Period'].unique()

array([2017, 2014, 2013, 2011, 2009, 2008, 2007, 2006, 2005, 2003, 2000,
       2016, 2015, 2012, 2010, 2004, 2002, 2001, 2018])

Data counts are all different for each year. This will need to be evaluated further since 2015 and 2016 might not be the best options.

In [21]:
ad_br_data[ad_br_data['Period'] == 2018].count()

Location         22
Period           22
Indicator        22
First Tooltip    22
dtype: int64

In [22]:
ad_br_data[ad_br_data['Period'] == 2017].count()

Location         99
Period           99
Indicator        99
First Tooltip    99
dtype: int64

In [23]:
ad_br_data[ad_br_data['Period'] == 2016].count()

Location         108
Period           108
Indicator        108
First Tooltip    108
dtype: int64

In [24]:
ad_br_data[ad_br_data['Period'] == 2015].count()

Location         106
Period           106
Indicator        106
First Tooltip    106
dtype: int64

In [25]:
ad_br_data[ad_br_data['Period'] == 2014].count()

Location         115
Period           115
Indicator        115
First Tooltip    115
dtype: int64

In [26]:
ad_br_data[ad_br_data['Period'] == 2013].count()

Location         116
Period           116
Indicator        116
First Tooltip    116
dtype: int64

In [27]:
ad_br_data[ad_br_data['Period'] == 2012].count()

Location         113
Period           113
Indicator        113
First Tooltip    113
dtype: int64

In [28]:
ad_br_data[ad_br_data['Period'] == 2011].count()

Location         117
Period           117
Indicator        117
First Tooltip    117
dtype: int64

In [29]:
ad_br_data[ad_br_data['Period'] == 2010].count()

Location         120
Period           120
Indicator        120
First Tooltip    120
dtype: int64

In [30]:
ad_br_data[ad_br_data['Period'] == 2009].count()

Location         133
Period           133
Indicator        133
First Tooltip    133
dtype: int64

In [31]:
ad_br_data[ad_br_data['Period'] == 2008].count()

Location         116
Period           116
Indicator        116
First Tooltip    116
dtype: int64

In [32]:
ad_br_data[ad_br_data['Period'] == 2007].count()

Location         121
Period           121
Indicator        121
First Tooltip    121
dtype: int64

In [33]:
ad_br_data[ad_br_data['Period'] == 2006].count()

Location         120
Period           120
Indicator        120
First Tooltip    120
dtype: int64

In [34]:
ad_br_data[ad_br_data['Period'] == 2005].count()

Location         130
Period           130
Indicator        130
First Tooltip    130
dtype: int64

In [35]:
ad_br_data[ad_br_data['Period'] == 2004].count()

Location         124
Period           124
Indicator        124
First Tooltip    124
dtype: int64

In [36]:
ad_br_data[ad_br_data['Period'] == 2003].count()

Location         120
Period           120
Indicator        120
First Tooltip    120
dtype: int64

In [37]:
ad_br_data[ad_br_data['Period'] == 2002].count()

Location         134
Period           134
Indicator        134
First Tooltip    134
dtype: int64

In [38]:
ad_br_data[ad_br_data['Period'] == 2001].count()

Location         133
Period           133
Indicator        133
First Tooltip    133
dtype: int64

In [39]:
ad_br_data[ad_br_data['Period'] == 2000].count()

Location         142
Period           142
Indicator        142
First Tooltip    142
dtype: int64

In [40]:
ad_br_data['Indicator'].unique()

array(['Adolescent birth rate (per 1000 women aged 15-19 years)'],
      dtype=object)

## Check if use this year

This year only has 108 rows

In [41]:
ad_br_2016 = ad_br_data[ad_br_data['Period'] == 2016]
ad_br_2016

Location  Period  \
12                                              Albania    2016   
41                                              Andorra    2016   
66                                  Antigua and Barbuda    2016   
96                                              Armenia    2016   
114                                           Australia    2016   
...                                                 ...     ...   
2065  United Kingdom of Great Britain and Northern I...    2016   
2082                        United Republic of Tanzania    2016   
2092                           United States of America    2016   
2111                                            Uruguay    2016   
2129                                         Uzbekistan    2016   

                                              Indicator  First Tooltip  
12    Adolescent birth rate (per 1000 women aged 15-...           17.3  
41    Adolescent birth rate (per 1000 women aged 15-...            2.6  
66    Adolescent birth rate (per 1000 women aged 15-...           31.1  
96    Adolescent birth rate (per 1000 women aged 15-...           24.5  
114   Adolescent birth rate (per 1000 women aged 15-...           10.3  
...                                                 ...            ...  
2065  Adolescent birth rate (per 1000 women aged 15-...           13.7  
2082  Adolescent birth rate (per 1000 women aged 15-...          138.9  
2092  Adolescent birth rate (per 1000 women aged 15-...           20.3  
2111  Adolescent birth rate (per 1000 women aged 15-...           50.3  
2129  Adolescent birth rate (per 1000 women aged 15-...           20.9  

[108 rows x 4 columns]

## Check if use this year

This row has only 106 rows

In [42]:
ad_br_2015 = ad_br_data[ad_br_data['Period'] == 2015]
ad_br_2015

Location  Period  \
13                                              Albania    2015   
28                                              Algeria    2015   
42                                              Andorra    2015   
67                                  Antigua and Barbuda    2015   
79                                            Argentina    2015   
...                                                 ...     ...   
2000                                             Turkey    2015   
2027                                             Uganda    2015   
2066  United Kingdom of Great Britain and Northern I...    2015   
2093                           United States of America    2015   
2112                                            Uruguay    2015   

                                              Indicator  First Tooltip  
13    Adolescent birth rate (per 1000 women aged 15-...           20.2  
28    Adolescent birth rate (per 1000 women aged 15-...            9.7  
42    Adolescent birth rate (per 1000 women aged 15-...            3.1  
67    Adolescent birth rate (per 1000 women aged 15-...           37.7  
79    Adolescent birth rate (per 1000 women aged 15-...           65.1  
...                                                 ...            ...  
2000  Adolescent birth rate (per 1000 women aged 15-...           24.9  
2027  Adolescent birth rate (per 1000 women aged 15-...          131.5  
2066  Adolescent birth rate (per 1000 women aged 15-...           14.4  
2093  Adolescent birth rate (per 1000 women aged 15-...           22.3  
2112  Adolescent birth rate (per 1000 women aged 15-...           55.6  

[106 rows x 4 columns]

## Air Pollution Deaths

In [43]:
ap_data = pd.read_csv(WHS_path/'airPollutionDeathRate.csv')
ap_data

Location                                   Dim2  \
0     Afghanistan                                  Total   
1     Afghanistan                                  Total   
2     Afghanistan                                  Total   
3     Afghanistan                                  Total   
4     Afghanistan                                  Total   
...           ...                                    ...   
6583     Zimbabwe  Chronic obstructive pulmonary disease   
6584     Zimbabwe  Chronic obstructive pulmonary disease   
6585     Zimbabwe  Chronic obstructive pulmonary disease   
6586     Zimbabwe  Chronic obstructive pulmonary disease   
6587     Zimbabwe  Chronic obstructive pulmonary disease   

                                              Indicator  Period        Dim1  \
0     Ambient and household air pollution attributab...    2016  Both sexes   
1     Ambient and household air pollution attributab...    2016        Male   
2     Ambient and household air pollution attributab...    2016      Female   
3     Ambient and household air pollution attributab...    2016  Both sexes   
4     Ambient and household air pollution attributab...    2016        Male   
...                                                 ...     ...         ...   
6583  Ambient and household air pollution attributab...    2016        Male   
6584  Ambient and household air pollution attributab...    2016      Female   
6585  Ambient and household air pollution attributab...    2016  Both sexes   
6586  Ambient and household air pollution attributab...    2016        Male   
6587  Ambient and household air pollution attributab...    2016      Female   

            First Tooltip  
0     95.12 [85.12-104.3]  
1        95.58 [85.5-106]  
2     94.63 [84.55-103.9]  
3     211.1 [188.9-231.3]  
4     228.7 [204.6-253.6]  
...                   ...  
6583     5.54 [2.74-7.66]  
6584      6.19 [4.73-7.3]  
6585  16.09 [11.67-19.89]  
6586   17.72 [8.75-24.51]  
6587  14.95 [11.43-17.62]  

[6588 rows x 6 columns]

In [44]:
ap_data.isnull().count()

Location         6588
Dim2             6588
Indicator        6588
Period           6588
Dim1             6588
First Tooltip    6588
dtype: int64

In [45]:
ap_data['Period'].unique()

array([2016])

In [46]:
ap_data['Dim2'].unique()

array(['Total', 'Lower respiratory infections',
       'Trachea, bronchus, lung cancers', 'Ischaemic heart disease',
       'Stroke', 'Chronic obstructive pulmonary disease'], dtype=object)

In [47]:
ap_data['Indicator'].unique()

array(['Ambient and household air pollution attributable death rate (per 100 000 population)',
       'Ambient and household air pollution attributable death rate (per 100 000 population, age-standardized)'],
      dtype=object)

I think it would be best to make each unique value in Dim2 as a column. That way we would have a single row for each country instead of a bunch of rows per country. Maybe see if the original dataset was set up like this.

In [48]:
ap_bs_data = ap_data[ap_data['Dim1'] == 'Both sexes']
ap_bs_data

Location                                   Dim2  \
0     Afghanistan                                  Total   
3     Afghanistan                                  Total   
6     Afghanistan           Lower respiratory infections   
9     Afghanistan           Lower respiratory infections   
12    Afghanistan        Trachea, bronchus, lung cancers   
...           ...                                    ...   
6573     Zimbabwe                Ischaemic heart disease   
6576     Zimbabwe                                 Stroke   
6579     Zimbabwe                                 Stroke   
6582     Zimbabwe  Chronic obstructive pulmonary disease   
6585     Zimbabwe  Chronic obstructive pulmonary disease   

                                              Indicator  Period        Dim1  \
0     Ambient and household air pollution attributab...    2016  Both sexes   
3     Ambient and household air pollution attributab...    2016  Both sexes   
6     Ambient and household air pollution attributab...    2016  Both sexes   
9     Ambient and household air pollution attributab...    2016  Both sexes   
12    Ambient and household air pollution attributab...    2016  Both sexes   
...                                                 ...     ...         ...   
6573  Ambient and household air pollution attributab...    2016  Both sexes   
6576  Ambient and household air pollution attributab...    2016  Both sexes   
6579  Ambient and household air pollution attributab...    2016  Both sexes   
6582  Ambient and household air pollution attributab...    2016  Both sexes   
6585  Ambient and household air pollution attributab...    2016  Both sexes   

            First Tooltip  
0     95.12 [85.12-104.3]  
3     211.1 [188.9-231.3]  
6     31.26 [25.66-35.67]  
9     31.68 [26.01-36.15]  
12       1.61 [1.21-1.96]  
...                   ...  
6573   33.24 [28.49-38.1]  
6576     8.27 [7.07-9.64]  
6579   20.08 [17.15-23.4]  
6582     5.88 [4.26-7.26]  
6585  16.09 [11.67-19.89]  

[2196 rows x 6 columns]

In [49]:
ap_bs_standardized = ap_bs_data[ap_bs_data['Indicator'] == 'Ambient and household air pollution attributable death rate (per 100 000 population, age-standardized)']
ap_bs_standardized

Location                                   Dim2  \
3     Afghanistan                                  Total   
9     Afghanistan           Lower respiratory infections   
15    Afghanistan        Trachea, bronchus, lung cancers   
21    Afghanistan                Ischaemic heart disease   
27    Afghanistan                                 Stroke   
...           ...                                    ...   
6561     Zimbabwe           Lower respiratory infections   
6567     Zimbabwe        Trachea, bronchus, lung cancers   
6573     Zimbabwe                Ischaemic heart disease   
6579     Zimbabwe                                 Stroke   
6585     Zimbabwe  Chronic obstructive pulmonary disease   

                                              Indicator  Period        Dim1  \
3     Ambient and household air pollution attributab...    2016  Both sexes   
9     Ambient and household air pollution attributab...    2016  Both sexes   
15    Ambient and household air pollution attributab...    2016  Both sexes   
21    Ambient and household air pollution attributab...    2016  Both sexes   
27    Ambient and household air pollution attributab...    2016  Both sexes   
...                                                 ...     ...         ...   
6561  Ambient and household air pollution attributab...    2016  Both sexes   
6567  Ambient and household air pollution attributab...    2016  Both sexes   
6573  Ambient and household air pollution attributab...    2016  Both sexes   
6579  Ambient and household air pollution attributab...    2016  Both sexes   
6585  Ambient and household air pollution attributab...    2016  Both sexes   

            First Tooltip  
3     211.1 [188.9-231.3]  
9     31.68 [26.01-36.15]  
15       4.05 [3.03-4.92]  
21    106.9 [91.91-121.1]  
27     45.1 [38.59-51.65]  
...                   ...  
6561  61.68 [51.05-71.05]  
6567     1.89 [1.42-2.33]  
6573   33.24 [28.49-38.1]  
6579   20.08 [17.15-23.4]  
6585  16.09 [11.67-19.89]  

[1098 rows x 6 columns]

In [50]:
ap_bs_standardized['Indicator'].unique()

array(['Ambient and household air pollution attributable death rate (per 100 000 population, age-standardized)'],
      dtype=object)

## This data has only 1 year. Maybe best to ignore.


In [159]:
ap_bs_tot = ap_bs_standardized[ap_bs_standardized['Dim2'] == 'Total']
ap_bs_tot

Location   Dim2  \
3                            Afghanistan  Total   
39                               Albania  Total   
75                               Algeria  Total   
111                               Angola  Total   
147                  Antigua and Barbuda  Total   
...                                  ...    ...   
6411  Venezuela (Bolivarian Republic of)  Total   
6447                            Viet Nam  Total   
6483                               Yemen  Total   
6519                              Zambia  Total   
6555                            Zimbabwe  Total   

                                              Indicator  Period        Dim1  \
3     Ambient and household air pollution attributab...    2016  Both sexes   
39    Ambient and household air pollution attributab...    2016  Both sexes   
75    Ambient and household air pollution attributab...    2016  Both sexes   
111   Ambient and household air pollution attributab...    2016  Both sexes   
147   Ambient and household air pollution attributab...    2016  Both sexes   
...                                                 ...     ...         ...   
6411  Ambient and household air pollution attributab...    2016  Both sexes   
6447  Ambient and household air pollution attributab...    2016  Both sexes   
6483  Ambient and household air pollution attributab...    2016  Both sexes   
6519  Ambient and household air pollution attributab...    2016  Both sexes   
6555  Ambient and household air pollution attributab...    2016  Both sexes   

            First Tooltip  
3     211.1 [188.9-231.3]  
39     67.99 [45.61-95.6]  
75    49.68 [38.61-86.22]  
111   118.5 [101.9-134.7]  
147    29.9 [25.18-35.55]  
...                   ...  
6411  34.65 [28.14-43.04]  
6447  64.48 [54.72-75.04]  
6483  194.2 [167.8-221.6]  
6519  127.2 [113.8-139.7]  
6555    133 [117.5-147.7]  

[183 rows x 6 columns]

## Alcohol Substance Abuse

In [51]:
alcohol_data = pd.read_csv(WHS_path/'alcoholSubstanceAbuse.csv')
alcohol_data

Location  Period                                          Indicator  \
0     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
1     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
2     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
3     Afghanistan    2015  Total (recorded+unrecorded) alcohol per capita...   
4     Afghanistan    2015  Total (recorded+unrecorded) alcohol per capita...   
...           ...     ...                                                ...   
2782     Zimbabwe    2005  Total (recorded+unrecorded) alcohol per capita...   
2783     Zimbabwe    2005  Total (recorded+unrecorded) alcohol per capita...   
2784     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   
2785     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   
2786     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   

            Dim1  First Tooltip  
0     Both sexes          0.210  
1           Male          0.360  
2         Female          0.055  
3     Both sexes          0.210  
4           Male          0.350  
...          ...            ...  
2782        Male          4.720  
2783      Female          0.990  
2784  Both sexes          2.460  
2785        Male          4.160  
2786      Female          0.880  

[2787 rows x 5 columns]

In [52]:
alcohol_data.isnull().count()

Location         2787
Period           2787
Indicator        2787
Dim1             2787
First Tooltip    2787
dtype: int64

In [53]:
alcohol_data['Period'].unique()

array([2018, 2015, 2010, 2005, 2000])

The years to use are 2018 and 2015.

In [54]:
alcohol_data[alcohol_data['Period'] == 2018].count()

Location         564
Period           564
Indicator        564
Dim1             564
First Tooltip    564
dtype: int64

In [55]:
alcohol_data[alcohol_data['Period'] == 2015].count()

Location         564
Period           564
Indicator        564
Dim1             564
First Tooltip    564
dtype: int64

In [56]:
alcohol_data[alcohol_data['Period'] == 2010].count()

Location         561
Period           561
Indicator        561
Dim1             561
First Tooltip    561
dtype: int64

In [57]:
alcohol_data[alcohol_data['Period'] == 2005].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [58]:
alcohol_data[alcohol_data['Period'] == 2000].count()

Location         549
Period           549
Indicator        549
Dim1             549
First Tooltip    549
dtype: int64

In [59]:
alcohol_data['Indicator'].unique()

array(['Total (recorded+unrecorded) alcohol per capita (15+) consumption'],
      dtype=object)

In [60]:
alcohol_both_sex = alcohol_data[alcohol_data['Dim1'] == 'Both sexes']
alcohol_both_sex

Location  Period                                          Indicator  \
0     Afghanistan    2018  Total (recorded+unrecorded) alcohol per capita...   
3     Afghanistan    2015  Total (recorded+unrecorded) alcohol per capita...   
6     Afghanistan    2010  Total (recorded+unrecorded) alcohol per capita...   
9         Albania    2018  Total (recorded+unrecorded) alcohol per capita...   
12        Albania    2015  Total (recorded+unrecorded) alcohol per capita...   
...           ...     ...                                                ...   
2772     Zimbabwe    2018  Total (recorded+unrecorded) alcohol per capita...   
2775     Zimbabwe    2015  Total (recorded+unrecorded) alcohol per capita...   
2778     Zimbabwe    2010  Total (recorded+unrecorded) alcohol per capita...   
2781     Zimbabwe    2005  Total (recorded+unrecorded) alcohol per capita...   
2784     Zimbabwe    2000  Total (recorded+unrecorded) alcohol per capita...   

            Dim1  First Tooltip  
0     Both sexes           0.21  
3     Both sexes           0.21  
6     Both sexes           0.21  
9     Both sexes           7.17  
12    Both sexes           6.74  
...          ...            ...  
2772  Both sexes           4.67  
2775  Both sexes           4.92  
2778  Both sexes           3.93  
2781  Both sexes           2.77  
2784  Both sexes           2.46  

[929 rows x 5 columns]

## Dataset to export

In [61]:
alcohol_both_sex_2018 = alcohol_both_sex[alcohol_both_sex['Period'] == 2018]
alcohol_both_sex_2018

Location  Period  \
0                            Afghanistan    2018   
9                                Albania    2018   
24                               Algeria    2018   
39                               Andorra    2018   
54                                Angola    2018   
...                                  ...     ...   
2712  Venezuela (Bolivarian Republic of)    2018   
2727                            Viet Nam    2018   
2742                               Yemen    2018   
2757                              Zambia    2018   
2772                            Zimbabwe    2018   

                                              Indicator        Dim1  \
0     Total (recorded+unrecorded) alcohol per capita...  Both sexes   
9     Total (recorded+unrecorded) alcohol per capita...  Both sexes   
24    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
39    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
54    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
...                                                 ...         ...   
2712  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2727  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2742  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2757  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2772  Total (recorded+unrecorded) alcohol per capita...  Both sexes   

      First Tooltip  
0             0.210  
9             7.170  
24            0.950  
39           11.020  
54            6.940  
...             ...  
2712          4.140  
2727          8.660  
2742          0.051  
2757          6.540  
2772          4.670  

[188 rows x 5 columns]

## Dataset to export

In [62]:
alcohol_both_sex_2015 = alcohol_both_sex[alcohol_both_sex['Period'] == 2015]
alcohol_both_sex_2015

Location  Period  \
3                            Afghanistan    2015   
12                               Albania    2015   
27                               Algeria    2015   
42                               Andorra    2015   
57                                Angola    2015   
...                                  ...     ...   
2715  Venezuela (Bolivarian Republic of)    2015   
2730                            Viet Nam    2015   
2745                               Yemen    2015   
2760                              Zambia    2015   
2775                            Zimbabwe    2015   

                                              Indicator        Dim1  \
3     Total (recorded+unrecorded) alcohol per capita...  Both sexes   
12    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
27    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
42    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
57    Total (recorded+unrecorded) alcohol per capita...  Both sexes   
...                                                 ...         ...   
2715  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2730  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2745  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2760  Total (recorded+unrecorded) alcohol per capita...  Both sexes   
2775  Total (recorded+unrecorded) alcohol per capita...  Both sexes   

      First Tooltip  
3             0.210  
12            6.740  
27            0.930  
42           11.010  
57            7.960  
...             ...  
2715          6.820  
2730          7.860  
2745          0.055  
2760          6.890  
2775          4.920  

[188 rows x 5 columns]

## Basic Sanitization

In [63]:
sani_data = pd.read_csv(WHS_path/'atLeastBasicSanitizationServices.csv')
sani_data

Location                                          Indicator  Period  \
0     Afghanistan  Population using at least basic sanitation ser...    2017   
1     Afghanistan  Population using at least basic sanitation ser...    2017   
2     Afghanistan  Population using at least basic sanitation ser...    2017   
3     Afghanistan  Population using at least basic sanitation ser...    2016   
4     Afghanistan  Population using at least basic sanitation ser...    2016   
...           ...                                                ...     ...   
9363     Zimbabwe  Population using at least basic sanitation ser...    2001   
9364     Zimbabwe  Population using at least basic sanitation ser...    2001   
9365     Zimbabwe  Population using at least basic sanitation ser...    2000   
9366     Zimbabwe  Population using at least basic sanitation ser...    2000   
9367     Zimbabwe  Population using at least basic sanitation ser...    2000   

       Dim1  First Tooltip  
0     Total          43.42  
1     Urban          62.26  
2     Rural          37.05  
3     Total          42.05  
4     Urban          59.92  
...     ...            ...  
9363  Urban          65.05  
9364  Rural          36.19  
9365  Total          45.93  
9366  Urban          65.05  
9367  Rural          36.19  

[9368 rows x 5 columns]

In [67]:
sani_data.isnull().count()

Location         9368
Indicator        9368
Period           9368
Dim1             9368
First Tooltip    9368
dtype: int64

The years 2013, 2012, and 2011 are slightly better than 2015 and 2016.

In [66]:
sani_data['Period'].unique()

array([2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 2007,
       2006, 2005, 2004, 2003, 2002, 2001, 2000])

In [69]:
sani_data[sani_data['Period'] == 2017].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [70]:
sani_data[sani_data['Period'] == 2016].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [71]:
sani_data[sani_data['Period'] == 2015].count()

Location         526
Indicator        526
Period           526
Dim1             526
First Tooltip    526
dtype: int64

In [72]:
sani_data[sani_data['Period'] == 2014].count()

Location         526
Indicator        526
Period           526
Dim1             526
First Tooltip    526
dtype: int64

In [73]:
sani_data[sani_data['Period'] == 2013].count()

Location         527
Indicator        527
Period           527
Dim1             527
First Tooltip    527
dtype: int64

In [74]:
sani_data[sani_data['Period'] == 2012].count()

Location         527
Indicator        527
Period           527
Dim1             527
First Tooltip    527
dtype: int64

In [75]:
sani_data[sani_data['Period'] == 2011].count()

Location         527
Indicator        527
Period           527
Dim1             527
First Tooltip    527
dtype: int64

In [76]:
sani_data[sani_data['Period'] == 2010].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [77]:
sani_data[sani_data['Period'] == 2009].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [78]:
sani_data[sani_data['Period'] == 2008].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [79]:
sani_data[sani_data['Period'] == 2007].count()

Location         524
Indicator        524
Period           524
Dim1             524
First Tooltip    524
dtype: int64

In [80]:
sani_data[sani_data['Period'] == 2006].count()

Location         523
Indicator        523
Period           523
Dim1             523
First Tooltip    523
dtype: int64

In [81]:
sani_data[sani_data['Period'] == 2005].count()

Location         517
Indicator        517
Period           517
Dim1             517
First Tooltip    517
dtype: int64

In [82]:
sani_data[sani_data['Period'] == 2004].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [83]:
sani_data[sani_data['Period'] == 2003].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [84]:
sani_data[sani_data['Period'] == 2002].count()

Location         516
Indicator        516
Period           516
Dim1             516
First Tooltip    516
dtype: int64

In [85]:
sani_data[sani_data['Period'] == 2001].count()

Location         510
Indicator        510
Period           510
Dim1             510
First Tooltip    510
dtype: int64

In [86]:
sani_data[sani_data['Period'] == 2000].count()

Location         501
Indicator        501
Period           501
Dim1             501
First Tooltip    501
dtype: int64

In [89]:
sani_data['Indicator'].unique()

array(['Population using at least basic sanitation services (%)'],
      dtype=object)

In [90]:
sani_data_tot = sani_data[sani_data['Dim1'] == "Total"]
sani_data_tot

Location                                          Indicator  Period  \
0     Afghanistan  Population using at least basic sanitation ser...    2017   
3     Afghanistan  Population using at least basic sanitation ser...    2016   
6     Afghanistan  Population using at least basic sanitation ser...    2015   
9     Afghanistan  Population using at least basic sanitation ser...    2014   
12    Afghanistan  Population using at least basic sanitation ser...    2013   
...           ...                                                ...     ...   
9353     Zimbabwe  Population using at least basic sanitation ser...    2004   
9356     Zimbabwe  Population using at least basic sanitation ser...    2003   
9359     Zimbabwe  Population using at least basic sanitation ser...    2002   
9362     Zimbabwe  Population using at least basic sanitation ser...    2001   
9365     Zimbabwe  Population using at least basic sanitation ser...    2000   

       Dim1  First Tooltip  
0     Total          43.42  
3     Total          42.05  
6     Total          40.71  
9     Total          39.37  
12    Total          38.06  
...     ...            ...  
9353  Total          45.37  
9356  Total          46.11  
9359  Total          46.15  
9362  Total          46.05  
9365  Total          45.93  

[3439 rows x 5 columns]

## Dataset to export

In [93]:
sani_data_2013 = sani_data_tot[sani_data_tot['Period'] == 2013]
sani_data_2013

Location  \
12                           Afghanistan   
66                               Albania   
120                              Algeria   
174                              Andorra   
228                               Angola   
...                                  ...   
9143  Venezuela (Bolivarian Republic of)   
9164                            Viet Nam   
9218                               Yemen   
9272                              Zambia   
9326                            Zimbabwe   

                                              Indicator  Period   Dim1  \
12    Population using at least basic sanitation ser...    2013  Total   
66    Population using at least basic sanitation ser...    2013  Total   
120   Population using at least basic sanitation ser...    2013  Total   
174   Population using at least basic sanitation ser...    2013  Total   
228   Population using at least basic sanitation ser...    2013  Total   
...                                                 ...     ...    ...   
9143  Population using at least basic sanitation ser...    2013  Total   
9164  Population using at least basic sanitation ser...    2013  Total   
9218  Population using at least basic sanitation ser...    2013  Total   
9272  Population using at least basic sanitation ser...    2013  Total   
9326  Population using at least basic sanitation ser...    2013  Total   

      First Tooltip  
12            38.06  
66            97.31  
120           87.16  
174          100.00  
228           44.85  
...             ...  
9143          93.94  
9164          76.49  
9218          56.31  
9272          25.74  
9326          38.94  

[194 rows x 5 columns]

## Dataset to export

In [94]:
sani_data_2012 = sani_data_tot[sani_data_tot['Period'] == 2012]
sani_data_2012

Location  \
15                           Afghanistan   
69                               Albania   
123                              Algeria   
177                              Andorra   
231                               Angola   
...                                  ...   
9144  Venezuela (Bolivarian Republic of)   
9167                            Viet Nam   
9221                               Yemen   
9275                              Zambia   
9329                            Zimbabwe   

                                              Indicator  Period   Dim1  \
15    Population using at least basic sanitation ser...    2012  Total   
69    Population using at least basic sanitation ser...    2012  Total   
123   Population using at least basic sanitation ser...    2012  Total   
177   Population using at least basic sanitation ser...    2012  Total   
231   Population using at least basic sanitation ser...    2012  Total   
...                                                 ...     ...    ...   
9144  Population using at least basic sanitation ser...    2012  Total   
9167  Population using at least basic sanitation ser...    2012  Total   
9221  Population using at least basic sanitation ser...    2012  Total   
9275  Population using at least basic sanitation ser...    2012  Total   
9329  Population using at least basic sanitation ser...    2012  Total   

      First Tooltip  
15            36.75  
69            96.78  
123           86.97  
177          100.00  
231           43.59  
...             ...  
9144          93.94  
9167          74.68  
9221          55.19  
9275          25.61  
9329          39.64  

[194 rows x 5 columns]

## Basic Drinking Water Services

In [95]:
water_data = pd.read_csv(WHS_path/'basicDrinkingWaterServices.csv')
water_data

Location  Period                                          Indicator  \
0     Afghanistan    2017  Population using at least basic drinking-water...   
1     Afghanistan    2016  Population using at least basic drinking-water...   
2     Afghanistan    2015  Population using at least basic drinking-water...   
3     Afghanistan    2014  Population using at least basic drinking-water...   
4     Afghanistan    2013  Population using at least basic drinking-water...   
...           ...     ...                                                ...   
3450     Zimbabwe    2004  Population using at least basic drinking-water...   
3451     Zimbabwe    2003  Population using at least basic drinking-water...   
3452     Zimbabwe    2002  Population using at least basic drinking-water...   
3453     Zimbabwe    2001  Population using at least basic drinking-water...   
3454     Zimbabwe    2000  Population using at least basic drinking-water...   

      First Tooltip  
0             57.32  
1             54.84  
2             52.39  
3             49.96  
4             47.56  
...             ...  
3450          57.94  
3451          58.59  
3452          59.23  
3453          59.88  
3454          59.88  

[3455 rows x 4 columns]

In [96]:
water_data.isnull().count()

Location         3455
Period           3455
Indicator        3455
First Tooltip    3455
dtype: int64

Year 2013 and 2012 are slightly better than 2015 and 2016

In [97]:
water_data['Period'].unique()

array([2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 2007,
       2006, 2005, 2004, 2003, 2002, 2001, 2000])

In [99]:
water_data[water_data['Period'] == 2017].count()

Location         190
Period           190
Indicator        190
First Tooltip    190
dtype: int64

In [100]:
water_data[water_data['Period'] == 2016].count()

Location         192
Period           192
Indicator        192
First Tooltip    192
dtype: int64

In [101]:
water_data[water_data['Period'] == 2015].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [102]:
water_data[water_data['Period'] == 2014].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [103]:
water_data[water_data['Period'] == 2013].count()

Location         194
Period           194
Indicator        194
First Tooltip    194
dtype: int64

In [104]:
water_data[water_data['Period'] == 2012].count()

Location         194
Period           194
Indicator        194
First Tooltip    194
dtype: int64

In [105]:
water_data[water_data['Period'] == 2011].count()

Location         194
Period           194
Indicator        194
First Tooltip    194
dtype: int64

In [106]:
water_data[water_data['Period'] == 2010].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [107]:
water_data[water_data['Period'] == 2009].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [108]:
water_data[water_data['Period'] == 2008].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [109]:
water_data[water_data['Period'] == 2007].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [110]:
water_data[water_data['Period'] == 2006].count()

Location         193
Period           193
Indicator        193
First Tooltip    193
dtype: int64

In [111]:
water_data[water_data['Period'] == 2005].count()

Location         192
Period           192
Indicator        192
First Tooltip    192
dtype: int64

In [112]:
water_data[water_data['Period'] == 2004].count()

Location         191
Period           191
Indicator        191
First Tooltip    191
dtype: int64

In [113]:
water_data[water_data['Period'] == 2003].count()

Location         191
Period           191
Indicator        191
First Tooltip    191
dtype: int64

In [114]:
water_data[water_data['Period'] == 2002].count()

Location         191
Period           191
Indicator        191
First Tooltip    191
dtype: int64

In [115]:
water_data[water_data['Period'] == 2001].count()

Location         189
Period           189
Indicator        189
First Tooltip    189
dtype: int64

In [116]:
water_data[water_data['Period'] == 2000].count()

Location         186
Period           186
Indicator        186
First Tooltip    186
dtype: int64

In [117]:
water_data['Indicator'].unique()

array(['Population using at least basic drinking-water services (%)'],
      dtype=object)

## Dataset to export

In [118]:
water_data_2013 = water_data[water_data['Period'] == 2013]
water_data_2013

Location  Period  \
4                            Afghanistan    2013   
22                               Albania    2013   
40                               Algeria    2013   
58                               Andorra    2013   
76                                Angola    2013   
...                                  ...     ...   
3369  Venezuela (Bolivarian Republic of)    2013   
3387                            Viet Nam    2013   
3405                               Yemen    2013   
3423                              Zambia    2013   
3441                            Zimbabwe    2013   

                                              Indicator  First Tooltip  
4     Population using at least basic drinking-water...          47.56  
22    Population using at least basic drinking-water...          86.23  
40    Population using at least basic drinking-water...          87.98  
58    Population using at least basic drinking-water...         100.00  
76    Population using at least basic drinking-water...          25.98  
...                                                 ...            ...  
3369  Population using at least basic drinking-water...          95.94  
3387  Population using at least basic drinking-water...          88.29  
3405  Population using at least basic drinking-water...          50.23  
3423  Population using at least basic drinking-water...          39.39  
3441  Population using at least basic drinking-water...          52.27  

[194 rows x 4 columns]

## Dataset to export

In [119]:
water_data_2012 = water_data[water_data['Period'] == 2012]
water_data_2012

Location  Period  \
5                            Afghanistan    2012   
23                               Albania    2012   
41                               Algeria    2012   
59                               Andorra    2012   
77                                Angola    2012   
...                                  ...     ...   
3370  Venezuela (Bolivarian Republic of)    2012   
3388                            Viet Nam    2012   
3406                               Yemen    2012   
3424                              Zambia    2012   
3442                            Zimbabwe    2012   

                                              Indicator  First Tooltip  
5     Population using at least basic drinking-water...          45.19  
23    Population using at least basic drinking-water...          82.59  
41    Population using at least basic drinking-water...          87.62  
59    Population using at least basic drinking-water...         100.00  
77    Population using at least basic drinking-water...          25.61  
...                                                 ...            ...  
3370  Population using at least basic drinking-water...          96.00  
3388  Population using at least basic drinking-water...          87.30  
3406  Population using at least basic drinking-water...          48.00  
3424  Population using at least basic drinking-water...          38.78  
3442  Population using at least basic drinking-water...          52.89  

[194 rows x 4 columns]

## Basic Hand Washing

In [120]:
hand_data = pd.read_csv(WHS_path/'basicHandWashing.csv')
hand_data

Location                                          Indicator  Period  \
0     Afghanistan  Population with basic handwashing facilities a...    2017   
1     Afghanistan  Population with basic handwashing facilities a...    2017   
2     Afghanistan  Population with basic handwashing facilities a...    2017   
3     Afghanistan  Population with basic handwashing facilities a...    2016   
4     Afghanistan  Population with basic handwashing facilities a...    2016   
...           ...                                                ...     ...   
2721     Zimbabwe  Population with basic handwashing facilities a...    2012   
2722     Zimbabwe  Population with basic handwashing facilities a...    2012   
2723     Zimbabwe  Population with basic handwashing facilities a...    2011   
2724     Zimbabwe  Population with basic handwashing facilities a...    2011   
2725     Zimbabwe  Population with basic handwashing facilities a...    2011   

       Dim1  First Tooltip  
0     Total          37.75  
1     Urban          63.58  
2     Rural          29.02  
3     Total          37.67  
4     Urban          63.58  
...     ...            ...  
2721  Urban          48.54  
2722  Rural          31.20  
2723  Total          36.93  
2724  Urban          48.54  
2725  Rural          31.20  

[2726 rows x 5 columns]

In [121]:
hand_data.isnull().count()

Location         2726
Indicator        2726
Period           2726
Dim1             2726
First Tooltip    2726
dtype: int64

Year 2014 and 2013 are looking the best. Better than 2016 and 2015.

In [122]:
hand_data['Period'].unique()

array([2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 2007,
       2006, 2005, 2004, 2003, 2002, 2001, 2000])

In [124]:
hand_data[hand_data['Period'] == 2017].count()

Location         231
Indicator        231
Period           231
Dim1             231
First Tooltip    231
dtype: int64

In [125]:
hand_data[hand_data['Period'] == 2016].count()

Location         247
Indicator        247
Period           247
Dim1             247
First Tooltip    247
dtype: int64

In [126]:
hand_data[hand_data['Period'] == 2015].count()

Location         265
Indicator        265
Period           265
Dim1             265
First Tooltip    265
dtype: int64

In [127]:
hand_data[hand_data['Period'] == 2014].count()

Location         277
Indicator        277
Period           277
Dim1             277
First Tooltip    277
dtype: int64

In [128]:
hand_data[hand_data['Period'] == 2013].count()

Location         274
Indicator        274
Period           274
Dim1             274
First Tooltip    274
dtype: int64

In [129]:
hand_data[hand_data['Period'] == 2012].count()

Location         252
Indicator        252
Period           252
Dim1             252
First Tooltip    252
dtype: int64

In [130]:
hand_data[hand_data['Period'] == 2011].count()

Location         225
Indicator        225
Period           225
Dim1             225
First Tooltip    225
dtype: int64

In [131]:
hand_data[hand_data['Period'] == 2010].count()

Location         192
Indicator        192
Period           192
Dim1             192
First Tooltip    192
dtype: int64

In [132]:
hand_data[hand_data['Period'] == 2009].count()

Location         159
Indicator        159
Period           159
Dim1             159
First Tooltip    159
dtype: int64

In [133]:
hand_data[hand_data['Period'] == 2008].count()

Location         139
Indicator        139
Period           139
Dim1             139
First Tooltip    139
dtype: int64

In [134]:
hand_data[hand_data['Period'] == 2007].count()

Location         120
Indicator        120
Period           120
Dim1             120
First Tooltip    120
dtype: int64

In [135]:
hand_data[hand_data['Period'] == 2006].count()

Location         105
Indicator        105
Period           105
Dim1             105
First Tooltip    105
dtype: int64

In [136]:
hand_data[hand_data['Period'] == 2005].count()

Location         72
Indicator        72
Period           72
Dim1             72
First Tooltip    72
dtype: int64

In [137]:
hand_data[hand_data['Period'] == 2004].count()

Location         54
Indicator        54
Period           54
Dim1             54
First Tooltip    54
dtype: int64

In [138]:
hand_data[hand_data['Period'] == 2003].count()

Location         30
Indicator        30
Period           30
Dim1             30
First Tooltip    30
dtype: int64

In [139]:
hand_data[hand_data['Period'] == 2002].count()

Location         30
Indicator        30
Period           30
Dim1             30
First Tooltip    30
dtype: int64

In [140]:
hand_data[hand_data['Period'] == 2001].count()

Location         27
Indicator        27
Period           27
Dim1             27
First Tooltip    27
dtype: int64

In [141]:
hand_data[hand_data['Period'] == 2000].count()

Location         27
Indicator        27
Period           27
Dim1             27
First Tooltip    27
dtype: int64

In [142]:
hand_data['Indicator'].unique()

array(['Population with basic handwashing facilities at home (%)'],
      dtype=object)

In [143]:
hand_data['Dim1'].unique()

array(['Total', 'Urban', 'Rural'], dtype=object)

In [145]:
hand_data_tot = hand_data[hand_data['Dim1'] == 'Total']
hand_data_tot

Location                                          Indicator  Period  \
0     Afghanistan  Population with basic handwashing facilities a...    2017   
3     Afghanistan  Population with basic handwashing facilities a...    2016   
6     Afghanistan  Population with basic handwashing facilities a...    2015   
9     Afghanistan  Population with basic handwashing facilities a...    2014   
12    Afghanistan  Population with basic handwashing facilities a...    2013   
...           ...                                                ...     ...   
2711     Zimbabwe  Population with basic handwashing facilities a...    2015   
2714     Zimbabwe  Population with basic handwashing facilities a...    2014   
2717     Zimbabwe  Population with basic handwashing facilities a...    2013   
2720     Zimbabwe  Population with basic handwashing facilities a...    2012   
2723     Zimbabwe  Population with basic handwashing facilities a...    2011   

       Dim1  First Tooltip  
0     Total          37.75  
3     Total          37.67  
6     Total          37.59  
9     Total          37.52  
12    Total          37.44  
...     ...            ...  
2711  Total          36.82  
2714  Total          36.84  
2717  Total          36.86  
2720  Total          36.89  
2723  Total          36.93  

[921 rows x 5 columns]

In [149]:
hand_data_tot[hand_data_tot['Period'] == 2017].count()

Location         78
Indicator        78
Period           78
Dim1             78
First Tooltip    78
dtype: int64

In [150]:
hand_data_tot[hand_data_tot['Period'] == 2016].count()

Location         84
Indicator        84
Period           84
Dim1             84
First Tooltip    84
dtype: int64

In [151]:
hand_data_tot[hand_data_tot['Period'] == 2015].count()

Location         90
Indicator        90
Period           90
Dim1             90
First Tooltip    90
dtype: int64

In [152]:
hand_data_tot[hand_data_tot['Period'] == 2014].count()

Location         94
Indicator        94
Period           94
Dim1             94
First Tooltip    94
dtype: int64

In [153]:
hand_data_tot[hand_data_tot['Period'] == 2013].count()

Location         93
Indicator        93
Period           93
Dim1             93
First Tooltip    93
dtype: int64

In [154]:
hand_data_tot[hand_data_tot['Period'] == 2012].count()

Location         85
Indicator        85
Period           85
Dim1             85
First Tooltip    85
dtype: int64

## Dataset to export

In [147]:
hand_data_2014 = hand_data_tot[hand_data_tot['Period'] == 2014]
hand_data_2014

Location                                          Indicator  Period  \
9     Afghanistan  Population with basic handwashing facilities a...    2014   
48        Algeria  Population with basic handwashing facilities a...    2014   
75         Angola  Population with basic handwashing facilities a...    2014   
93        Armenia  Population with basic handwashing facilities a...    2014   
141    Azerbaijan  Population with basic handwashing facilities a...    2014   
...           ...                                                ...     ...   
2597      Vanuatu  Population with basic handwashing facilities a...    2014   
2624     Viet Nam  Population with basic handwashing facilities a...    2014   
2663        Yemen  Population with basic handwashing facilities a...    2014   
2690       Zambia  Population with basic handwashing facilities a...    2014   
2714     Zimbabwe  Population with basic handwashing facilities a...    2014   

       Dim1  First Tooltip  
9     Total          37.52  
48    Total          83.47  
75    Total          26.23  
93    Total          92.06  
141   Total          83.24  
...     ...            ...  
2597  Total          25.12  
2624  Total          85.63  
2663  Total          48.93  
2690  Total          13.61  
2714  Total          36.84  

[94 rows x 5 columns]

## Dataset to export

In [148]:
hand_data_2013 = hand_data_tot[hand_data_tot['Period'] == 2013]
hand_data_2013

Location                                          Indicator  Period  \
12    Afghanistan  Population with basic handwashing facilities a...    2013   
51        Algeria  Population with basic handwashing facilities a...    2013   
78         Angola  Population with basic handwashing facilities a...    2013   
96        Armenia  Population with basic handwashing facilities a...    2013   
142    Azerbaijan  Population with basic handwashing facilities a...    2013   
...           ...                                                ...     ...   
2600      Vanuatu  Population with basic handwashing facilities a...    2013   
2627     Viet Nam  Population with basic handwashing facilities a...    2013   
2666        Yemen  Population with basic handwashing facilities a...    2013   
2693       Zambia  Population with basic handwashing facilities a...    2013   
2717     Zimbabwe  Population with basic handwashing facilities a...    2013   

       Dim1  First Tooltip  
12    Total          37.44  
51    Total          83.37  
78    Total          26.08  
96    Total          91.41  
142   Total          83.24  
...     ...            ...  
2600  Total          25.09  
2627  Total          85.56  
2666  Total          48.72  
2693  Total          13.51  
2717  Total          36.86  

[93 rows x 5 columns]

## Birth Attended by Skilled Personnel

In [155]:
basp_data = pd.read_csv(WHS_path/'birthAttendedBySkilledPersonal.csv')
basp_data

Location  Period                                        Indicator  \
0     Afghanistan    2018  Births attended by skilled health personnel (%)   
1     Afghanistan    2017  Births attended by skilled health personnel (%)   
2     Afghanistan    2015  Births attended by skilled health personnel (%)   
3     Afghanistan    2014  Births attended by skilled health personnel (%)   
4     Afghanistan    2012  Births attended by skilled health personnel (%)   
...           ...     ...                                              ...   
1750     Zimbabwe    2015  Births attended by skilled health personnel (%)   
1751     Zimbabwe    2014  Births attended by skilled health personnel (%)   
1752     Zimbabwe    2011  Births attended by skilled health personnel (%)   
1753     Zimbabwe    2009  Births attended by skilled health personnel (%)   
1754     Zimbabwe    2006  Births attended by skilled health personnel (%)   

      First Tooltip  
0              58.8  
1              53.4  
2              50.5  
3              45.2  
4              39.9  
...             ...  
1750           78.1  
1751           80.0  
1752           66.2  
1753           60.2  
1754           68.5  

[1755 rows x 4 columns]

In [156]:
basp_data.isnull().count()

Location         1755
Period           1755
Indicator        1755
First Tooltip    1755
dtype: int64

In [157]:
basp_data['Indicator'].unique()

array(['Births attended by skilled health personnel (%)'], dtype=object)

In [184]:
basp_data[basp_data['First Tooltip'] == 0.0].count()

Location         0
Period           0
Indicator        0
First Tooltip    0
dtype: int64

The best aren't next to each other. They are 2014, 2010, and 2006. Of those next to each other, 2006 and 2007 are the winners.

In [160]:
basp_data['Period'].unique()

array([2018, 2017, 2015, 2014, 2012, 2011, 2010, 2003, 2009, 2005, 2013,
       2006, 2016, 2007, 2008, 2004, 2001, 2000, 2002, 2019])

In [161]:
basp_data[basp_data['Period'] == 2019].count()

Location         6
Period           6
Indicator        6
First Tooltip    6
dtype: int64

In [162]:
basp_data[basp_data['Period'] == 2018].count()

Location         60
Period           60
Indicator        60
First Tooltip    60
dtype: int64

In [163]:
basp_data[basp_data['Period'] == 2017].count()

Location         81
Period           81
Indicator        81
First Tooltip    81
dtype: int64

In [164]:
basp_data[basp_data['Period'] == 2016].count()

Location         90
Period           90
Indicator        90
First Tooltip    90
dtype: int64

In [165]:
basp_data[basp_data['Period'] == 2015].count()

Location         94
Period           94
Indicator        94
First Tooltip    94
dtype: int64

In [166]:
basp_data[basp_data['Period'] == 2014].count()

Location         112
Period           112
Indicator        112
First Tooltip    112
dtype: int64

In [167]:
basp_data[basp_data['Period'] == 2013].count()

Location         98
Period           98
Indicator        98
First Tooltip    98
dtype: int64

In [168]:
basp_data[basp_data['Period'] == 2012].count()

Location         104
Period           104
Indicator        104
First Tooltip    104
dtype: int64

In [169]:
basp_data[basp_data['Period'] == 2011].count()

Location         103
Period           103
Indicator        103
First Tooltip    103
dtype: int64

In [170]:
basp_data[basp_data['Period'] == 2010].count()

Location         113
Period           113
Indicator        113
First Tooltip    113
dtype: int64

In [171]:
basp_data[basp_data['Period'] == 2009].count()

Location         102
Period           102
Indicator        102
First Tooltip    102
dtype: int64

In [172]:
basp_data[basp_data['Period'] == 2008].count()

Location         99
Period           99
Indicator        99
First Tooltip    99
dtype: int64

In [173]:
basp_data[basp_data['Period'] == 2007].count()

Location         107
Period           107
Indicator        107
First Tooltip    107
dtype: int64

In [174]:
basp_data[basp_data['Period'] == 2006].count()

Location         118
Period           118
Indicator        118
First Tooltip    118
dtype: int64

In [175]:
basp_data[basp_data['Period'] == 2005].count()

Location         99
Period           99
Indicator        99
First Tooltip    99
dtype: int64

In [176]:
basp_data[basp_data['Period'] == 2004].count()

Location         78
Period           78
Indicator        78
First Tooltip    78
dtype: int64

In [177]:
basp_data[basp_data['Period'] == 2003].count()

Location         77
Period           77
Indicator        77
First Tooltip    77
dtype: int64

In [178]:
basp_data[basp_data['Period'] == 2002].count()

Location         75
Period           75
Indicator        75
First Tooltip    75
dtype: int64

In [179]:
basp_data[basp_data['Period'] == 2001].count()

Location         63
Period           63
Indicator        63
First Tooltip    63
dtype: int64

In [180]:
basp_data[basp_data['Period'] == 2000].count()

Location         76
Period           76
Indicator        76
First Tooltip    76
dtype: int64

## Dataset to export

In [197]:
basp_data_2007 = basp_data[basp_data['Period'] == 2007]
basp_data_2007

Location  Period  \
17                      Angola    2007   
26         Antigua and Barbuda    2007   
43                   Argentina    2007   
59                     Armenia    2007   
76                   Australia    2007   
...                        ...     ...   
1684  United States of America    2007   
1702                   Uruguay    2007   
1721                Uzbekistan    2007   
1729                   Vanuatu    2007   
1748                    Zambia    2007   

                                            Indicator  First Tooltip  
17    Births attended by skilled health personnel (%)           47.3  
26    Births attended by skilled health personnel (%)          100.0  
43    Births attended by skilled health personnel (%)           99.4  
59    Births attended by skilled health personnel (%)           99.9  
76    Births attended by skilled health personnel (%)           99.2  
...                                               ...            ...  
1684  Births attended by skilled health personnel (%)           99.3  
1702  Births attended by skilled health personnel (%)           98.6  
1721  Births attended by skilled health personnel (%)          100.0  
1729  Births attended by skilled health personnel (%)           74.0  
1748  Births attended by skilled health personnel (%)           45.2  

[107 rows x 4 columns]

## Dataset to export

In [198]:
basp_data_2006= basp_data[basp_data['Period'] == 2006]
basp_data_2006

Location  Period  \
12                Algeria    2006   
27    Antigua and Barbuda    2006   
44              Argentina    2006   
60                Armenia    2006   
77              Australia    2006   
...                   ...     ...   
1722           Uzbekistan    2006   
1730              Vanuatu    2006   
1743             Viet Nam    2006   
1746                Yemen    2006   
1754             Zimbabwe    2006   

                                            Indicator  First Tooltip  
12    Births attended by skilled health personnel (%)           95.2  
27    Births attended by skilled health personnel (%)           99.9  
44    Births attended by skilled health personnel (%)           99.4  
60    Births attended by skilled health personnel (%)           99.7  
77    Births attended by skilled health personnel (%)           99.3  
...                                               ...            ...  
1722  Births attended by skilled health personnel (%)           99.6  
1730  Births attended by skilled health personnel (%)           92.9  
1743  Births attended by skilled health personnel (%)           83.8  
1746  Births attended by skilled health personnel (%)           35.7  
1754  Births attended by skilled health personnel (%)           68.5  

[118 rows x 4 columns]

## Clean Fuel and Tech - IGNORE

## Crude Suicide Rates - IGNORE

## Data Availability for UHC - IGNORE

## Dentists

In [200]:
dentist_data = pd.read_csv(WHS_path/'dentists.csv')
dentist_data

Location  Period              Indicator  First Tooltip
0     Afghanistan    2016  Dentists (per 10,000)          0.034
1     Afghanistan    2015  Dentists (per 10,000)          0.036
2     Afghanistan    2014  Dentists (per 10,000)          0.033
3     Afghanistan    2008  Dentists (per 10,000)          0.140
4     Afghanistan    2007  Dentists (per 10,000)          0.120
...           ...     ...                    ...            ...
2129     Zimbabwe    2009  Dentists (per 10,000)          0.150
2130     Zimbabwe    2008  Dentists (per 10,000)          0.150
2131     Zimbabwe    2007  Dentists (per 10,000)          0.160
2132     Zimbabwe    1995  Dentists (per 10,000)          0.130
2133     Zimbabwe    1990  Dentists (per 10,000)          0.130

[2134 rows x 4 columns]

In [202]:
dentist_data.isnull().count()

Location         2134
Period           2134
Indicator        2134
First Tooltip    2134
dtype: int64

In [255]:
dentist_data[dentist_data['First Tooltip'] == 0.0]

Empty DataFrame
Columns: [Location, Period, Indicator, First Tooltip]
Index: []

In [201]:
dentist_data['Indicator'].unique()

array(['Dentists (per 10,000)'], dtype=object)

I ignored the data from the 90s because I don't have time for that. Best consecutive datasets are 2009 and 2010.

In [204]:
dentist_data['Period'].unique()

array([2016, 2015, 2014, 2008, 2007, 2006, 2003, 2002, 2001, 2000, 1999,
       1998, 1997, 1996, 1995, 1994, 1993, 1992, 1991, 1990, 2018, 2005,
       2009, 2004, 2017, 2013, 2012, 2011, 2010, 2019])

In [206]:
dentist_data[dentist_data['Period'] == 2019].count()

Location         32
Period           32
Indicator        32
First Tooltip    32
dtype: int64

In [207]:
dentist_data[dentist_data['Period'] == 2018].count()

Location         63
Period           63
Indicator        63
First Tooltip    63
dtype: int64

In [208]:
dentist_data[dentist_data['Period'] == 2017].count()

Location         80
Period           80
Indicator        80
First Tooltip    80
dtype: int64

In [210]:
dentist_data[dentist_data['Period'] == 2016].count()

Location         83
Period           83
Indicator        83
First Tooltip    83
dtype: int64

In [211]:
dentist_data[dentist_data['Period'] == 2015].count()

Location         87
Period           87
Indicator        87
First Tooltip    87
dtype: int64

In [212]:
dentist_data[dentist_data['Period'] == 2014].count()

Location         92
Period           92
Indicator        92
First Tooltip    92
dtype: int64

In [213]:
dentist_data[dentist_data['Period'] == 2013].count()

Location         85
Period           85
Indicator        85
First Tooltip    85
dtype: int64

In [214]:
dentist_data[dentist_data['Period'] == 2012].count()

Location         93
Period           93
Indicator        93
First Tooltip    93
dtype: int64

In [215]:
dentist_data[dentist_data['Period'] == 2011].count()

Location         97
Period           97
Indicator        97
First Tooltip    97
dtype: int64

In [216]:
dentist_data[dentist_data['Period'] == 2010].count()

Location         108
Period           108
Indicator        108
First Tooltip    108
dtype: int64

In [217]:
dentist_data[dentist_data['Period'] == 2009].count()

Location         114
Period           114
Indicator        114
First Tooltip    114
dtype: int64

In [218]:
dentist_data[dentist_data['Period'] == 2008].count()

Location         110
Period           110
Indicator        110
First Tooltip    110
dtype: int64

In [219]:
dentist_data[dentist_data['Period'] == 2007].count()

Location         88
Period           88
Indicator        88
First Tooltip    88
dtype: int64

In [220]:
dentist_data[dentist_data['Period'] == 2006].count()

Location         80
Period           80
Indicator        80
First Tooltip    80
dtype: int64

In [221]:
dentist_data[dentist_data['Period'] == 2005].count()

Location         82
Period           82
Indicator        82
First Tooltip    82
dtype: int64

In [222]:
dentist_data[dentist_data['Period'] == 2004].count()

Location         113
Period           113
Indicator        113
First Tooltip    113
dtype: int64

In [223]:
dentist_data[dentist_data['Period'] == 2003].count()

Location         70
Period           70
Indicator        70
First Tooltip    70
dtype: int64

In [224]:
dentist_data[dentist_data['Period'] == 2002].count()

Location         73
Period           73
Indicator        73
First Tooltip    73
dtype: int64

In [225]:
dentist_data[dentist_data['Period'] == 2001].count()

Location         70
Period           70
Indicator        70
First Tooltip    70
dtype: int64

In [226]:
dentist_data[dentist_data['Period'] == 2000].count()

Location         74
Period           74
Indicator        74
First Tooltip    74
dtype: int64

## Dataset to export

In [227]:
d_data_2010 = dentist_data[dentist_data['Period'] == 2010]
d_data_2010

Location  Period              Indicator  First Tooltip
53                     Armenia    2010  Dentists (per 10,000)          4.370
71                   Australia    2010  Dentists (per 10,000)          5.180
84                     Austria    2010  Dentists (per 10,000)          5.570
109                 Azerbaijan    2010  Dentists (per 10,000)          2.730
138                    Bahrain    2010  Dentists (per 10,000)          0.980
...                        ...     ...                    ...            ...
2080  United States of America    2010  Dentists (per 10,000)          5.780
2091                Uzbekistan    2010  Dentists (per 10,000)          1.650
2110                     Yemen    2010  Dentists (per 10,000)          0.250
2117                    Zambia    2010  Dentists (per 10,000)          0.180
2128                  Zimbabwe    2010  Dentists (per 10,000)          0.035

[108 rows x 4 columns]

## Dataset to export

In [228]:
d_data_2009 = dentist_data[dentist_data['Period'] == 2009]
d_data_2009

Location  Period  \
26                                              Andorra    2009   
54                                              Armenia    2009   
72                                            Australia    2009   
85                                              Austria    2009   
110                                          Azerbaijan    2009   
...                                                 ...     ...   
2052                               United Arab Emirates    2009   
2071  United Kingdom of Great Britain and Northern I...    2009   
2092                                         Uzbekistan    2009   
2118                                             Zambia    2009   
2129                                           Zimbabwe    2009   

                  Indicator  First Tooltip  
26    Dentists (per 10,000)           6.04  
54    Dentists (per 10,000)           4.31  
72    Dentists (per 10,000)           5.27  
85    Dentists (per 10,000)           5.52  
110   Dentists (per 10,000)           2.75  
...                     ...            ...  
2052  Dentists (per 10,000)           3.63  
2071  Dentists (per 10,000)           5.00  
2092  Dentists (per 10,000)           1.59  
2118  Dentists (per 10,000)           0.18  
2129  Dentists (per 10,000)           0.15  

[114 rows x 4 columns]

## Violence Against Women - IGNORE

## Healthy Life Expectancy at Birth

In [231]:
hleb_data = pd.read_csv(WHS_path/'HALElifeExpectancyAtBirth.csv')
hleb_data

Location  Period                                        Indicator  \
0     Afghanistan    2019  Healthy life expectancy (HALE) at birth (years)   
1     Afghanistan    2019  Healthy life expectancy (HALE) at birth (years)   
2     Afghanistan    2019  Healthy life expectancy (HALE) at birth (years)   
3     Afghanistan    2015  Healthy life expectancy (HALE) at birth (years)   
4     Afghanistan    2015  Healthy life expectancy (HALE) at birth (years)   
...           ...     ...                                              ...   
2191     Zimbabwe    2010  Healthy life expectancy (HALE) at birth (years)   
2192     Zimbabwe    2010  Healthy life expectancy (HALE) at birth (years)   
2193     Zimbabwe    2000  Healthy life expectancy (HALE) at birth (years)   
2194     Zimbabwe    2000  Healthy life expectancy (HALE) at birth (years)   
2195     Zimbabwe    2000  Healthy life expectancy (HALE) at birth (years)   

            Dim1  First Tooltip  
0     Both sexes          53.95  
1           Male          54.73  
2         Female          53.15  
3     Both sexes          52.60  
4           Male          52.78  
...          ...            ...  
2191        Male          44.25  
2192      Female          46.12  
2193  Both sexes          41.03  
2194        Male          40.39  
2195      Female          41.77  

[2196 rows x 5 columns]

In [232]:
hleb_data.isnull().count()

Location         2196
Period           2196
Indicator        2196
Dim1             2196
First Tooltip    2196
dtype: int64

In [233]:
hleb_data[hleb_data['First Tooltip'] == 0.0]

Empty DataFrame
Columns: [Location, Period, Indicator, Dim1, First Tooltip]
Index: []

In [234]:
hleb_data['Indicator'].unique()

array(['Healthy life expectancy (HALE) at birth (years)'], dtype=object)

In [238]:
hleb_bs = hleb_data[hleb_data['Dim1'] == "Both sexes"]
hleb_bs

Location  Period                                        Indicator  \
0     Afghanistan    2019  Healthy life expectancy (HALE) at birth (years)   
3     Afghanistan    2015  Healthy life expectancy (HALE) at birth (years)   
6     Afghanistan    2010  Healthy life expectancy (HALE) at birth (years)   
9     Afghanistan    2000  Healthy life expectancy (HALE) at birth (years)   
12        Albania    2019  Healthy life expectancy (HALE) at birth (years)   
...           ...     ...                                              ...   
2181       Zambia    2000  Healthy life expectancy (HALE) at birth (years)   
2184     Zimbabwe    2019  Healthy life expectancy (HALE) at birth (years)   
2187     Zimbabwe    2015  Healthy life expectancy (HALE) at birth (years)   
2190     Zimbabwe    2010  Healthy life expectancy (HALE) at birth (years)   
2193     Zimbabwe    2000  Healthy life expectancy (HALE) at birth (years)   

            Dim1  First Tooltip  
0     Both sexes          53.95  
3     Both sexes          52.60  
6     Both sexes          51.13  
9     Both sexes          46.76  
12    Both sexes          69.08  
...          ...            ...  
2181  Both sexes          38.95  
2184  Both sexes          53.07  
2187  Both sexes          51.17  
2190  Both sexes          45.22  
2193  Both sexes          41.03  

[732 rows x 5 columns]

All years have the same number! A great number too. For that reason, I'll select the most recent: 2019 and 2015.

In [239]:
hleb_bs['Period'].unique()

array([2019, 2015, 2010, 2000])

In [240]:
hleb_bs[hleb_bs['Period'] == 2019].count()

Location         183
Period           183
Indicator        183
Dim1             183
First Tooltip    183
dtype: int64

In [241]:
hleb_bs[hleb_bs['Period'] == 2015].count()

Location         183
Period           183
Indicator        183
Dim1             183
First Tooltip    183
dtype: int64

In [242]:
hleb_bs[hleb_bs['Period'] == 2010].count()

Location         183
Period           183
Indicator        183
Dim1             183
First Tooltip    183
dtype: int64

In [243]:
hleb_bs[hleb_bs['Period'] == 2000].count()

Location         183
Period           183
Indicator        183
Dim1             183
First Tooltip    183
dtype: int64

## Dataset to export

In [244]:
hleb_2019 = hleb_bs[hleb_bs['Period'] == 2019]
hleb_2019

Location  Period  \
0                            Afghanistan    2019   
12                               Albania    2019   
24                               Algeria    2019   
36                                Angola    2019   
48                   Antigua and Barbuda    2019   
...                                  ...     ...   
2136  Venezuela (Bolivarian Republic of)    2019   
2148                            Viet Nam    2019   
2160                               Yemen    2019   
2172                              Zambia    2019   
2184                            Zimbabwe    2019   

                                            Indicator        Dim1  \
0     Healthy life expectancy (HALE) at birth (years)  Both sexes   
12    Healthy life expectancy (HALE) at birth (years)  Both sexes   
24    Healthy life expectancy (HALE) at birth (years)  Both sexes   
36    Healthy life expectancy (HALE) at birth (years)  Both sexes   
48    Healthy life expectancy (HALE) at birth (years)  Both sexes   
...                                               ...         ...   
2136  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2148  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2160  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2172  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2184  Healthy life expectancy (HALE) at birth (years)  Both sexes   

      First Tooltip  
0             53.95  
12            69.08  
24            66.39  
36            54.84  
48            66.97  
...             ...  
2136          64.44  
2148          65.30  
2160          57.53  
2172          54.40  
2184          53.07  

[183 rows x 5 columns]

## Dataset to export

In [245]:
hleb_2015 = hleb_bs[hleb_bs['Period'] == 2015]
hleb_2015

Location  Period  \
3                            Afghanistan    2015   
15                               Albania    2015   
27                               Algeria    2015   
39                                Angola    2015   
51                   Antigua and Barbuda    2015   
...                                  ...     ...   
2139  Venezuela (Bolivarian Republic of)    2015   
2151                            Viet Nam    2015   
2163                               Yemen    2015   
2175                              Zambia    2015   
2187                            Zimbabwe    2015   

                                            Indicator        Dim1  \
3     Healthy life expectancy (HALE) at birth (years)  Both sexes   
15    Healthy life expectancy (HALE) at birth (years)  Both sexes   
27    Healthy life expectancy (HALE) at birth (years)  Both sexes   
39    Healthy life expectancy (HALE) at birth (years)  Both sexes   
51    Healthy life expectancy (HALE) at birth (years)  Both sexes   
...                                               ...         ...   
2139  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2151  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2163  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2175  Healthy life expectancy (HALE) at birth (years)  Both sexes   
2187  Healthy life expectancy (HALE) at birth (years)  Both sexes   

      First Tooltip  
3             52.60  
15            69.03  
27            65.99  
39            53.66  
51            66.81  
...             ...  
2139          65.10  
2151          64.89  
2163          58.38  
2175          52.71  
2187          51.17  

[183 rows x 5 columns]

## Healthy Life Expectancy at Birth WHO Region - IGNORE

## Hepatitis B Surface Antigen

In [247]:
hepb_data = pd.read_csv(WHS_path/'hepatitusBsurfaceAntigen.csv')
hepb_data

Location  \
0                           Afghanistan   
1                               Albania   
2                               Algeria   
3                               Andorra   
4                                Angola   
..                                  ...   
189  Venezuela (Bolivarian Republic of)   
190                            Viet Nam   
191                               Yemen   
192                              Zambia   
193                            Zimbabwe   

                                             Indicator  Period  \
0    Hepatitis B surface antigen (HBsAg) prevalence...    2015   
1    Hepatitis B surface antigen (HBsAg) prevalence...    2015   
2    Hepatitis B surface antigen (HBsAg) prevalence...    2015   
3    Hepatitis B surface antigen (HBsAg) prevalence...    2015   
4    Hepatitis B surface antigen (HBsAg) prevalence...    2015   
..                                                 ...     ...   
189  Hepatitis B surface antigen (HBsAg) prevalence...    2015   
190  Hepatitis B surface antigen (HBsAg) prevalence...    2015   
191  Hepatitis B surface antigen (HBsAg) prevalence...    2015   
192  Hepatitis B surface antigen (HBsAg) prevalence...    2015   
193  Hepatitis B surface antigen (HBsAg) prevalence...    2015   

        First Tooltip  
0     0.5 [0.29-0.78]  
1    1.29 [0.83-1.91]  
2    0.22 [0.12-0.34]  
3       0.08 [0-1.65]  
4    4.85 [3.29-6.81]  
..                ...  
189  0.62 [0.41-0.91]  
190   1.2 [0.84-1.62]  
191  2.54 [1.82-3.54]  
192  1.84 [1.17-2.76]  
193  4.38 [2.92-6.33]  

[194 rows x 4 columns]

In [248]:
hepb_data.isnull().count()

Location         194
Indicator        194
Period           194
First Tooltip    194
dtype: int64

In [249]:
hepb_data[hepb_data['First Tooltip'] == 0.0]

Empty DataFrame
Columns: [Location, Indicator, Period, First Tooltip]
Index: []

In [250]:
hepb_data['Indicator'].unique()

array(['Hepatitis B surface antigen (HBsAg) prevalence among children under 5 years (%)'],
      dtype=object)

The only year is in 2015

In [251]:
hepb_data['Period'].unique()

array([2015])

## HepB data only has 1 year. May be best to ignore.

## Malaria

In [256]:
malaria_data = pd.read_csv(WHS_path/'incedenceOfMalaria.csv')
malaria_data

Location                                         Indicator  Period  \
0     Afghanistan  Malaria incidence (per 1 000 population at risk)    2018   
1     Afghanistan  Malaria incidence (per 1 000 population at risk)    2017   
2     Afghanistan  Malaria incidence (per 1 000 population at risk)    2016   
3     Afghanistan  Malaria incidence (per 1 000 population at risk)    2015   
4     Afghanistan  Malaria incidence (per 1 000 population at risk)    2014   
...           ...                                               ...     ...   
2028     Zimbabwe  Malaria incidence (per 1 000 population at risk)    2004   
2029     Zimbabwe  Malaria incidence (per 1 000 population at risk)    2003   
2030     Zimbabwe  Malaria incidence (per 1 000 population at risk)    2002   
2031     Zimbabwe  Malaria incidence (per 1 000 population at risk)    2001   
2032     Zimbabwe  Malaria incidence (per 1 000 population at risk)    2000   

      First Tooltip  
0             29.01  
1             27.07  
2             26.11  
3             14.91  
4             12.67  
...             ...  
2028         120.20  
2029         120.90  
2030         121.50  
2031         122.10  
2032         123.00  

[2033 rows x 4 columns]

In [257]:
malaria_data.isnull().count()

Location         2033
Indicator        2033
Period           2033
First Tooltip    2033
dtype: int64

This dataset has a high number of zeros. The data has a chance of being accurrate or inaccurate. I guess it depends if mosquitos can be found in that country.

In [261]:
malaria_data[malaria_data['First Tooltip'] == 0.0]

Location                                         Indicator  Period  \
19       Algeria  Malaria incidence (per 1 000 population at risk)    2018   
20       Algeria  Malaria incidence (per 1 000 population at risk)    2017   
21       Algeria  Malaria incidence (per 1 000 population at risk)    2016   
22       Algeria  Malaria incidence (per 1 000 population at risk)    2015   
23       Algeria  Malaria incidence (per 1 000 population at risk)    2014   
...          ...                                               ...     ...   
1904  Uzbekistan  Malaria incidence (per 1 000 population at risk)    2014   
1905  Uzbekistan  Malaria incidence (per 1 000 population at risk)    2013   
1906  Uzbekistan  Malaria incidence (per 1 000 population at risk)    2012   
1907  Uzbekistan  Malaria incidence (per 1 000 population at risk)    2011   
1909  Uzbekistan  Malaria incidence (per 1 000 population at risk)    2009   

      First Tooltip  
19              0.0  
20              0.0  
21              0.0  
22              0.0  
23              0.0  
...             ...  
1904            0.0  
1905            0.0  
1906            0.0  
1907            0.0  
1909            0.0  

[221 rows x 4 columns]

In [262]:
malaria_data['Indicator'].unique()

array(['Malaria incidence (per 1 000 population at risk)'], dtype=object)

All years have the exact same number at 107! For that reason, I'm just selecting 2015 and 2016.

In [263]:
malaria_data['Period'].unique()

array([2018, 2017, 2016, 2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008,
       2007, 2006, 2005, 2004, 2003, 2002, 2001, 2000])

In [265]:
malaria_data[malaria_data['Period'] == 2018].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [266]:
malaria_data[malaria_data['Period'] == 2017].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [267]:
malaria_data[malaria_data['Period'] == 2016].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [268]:
malaria_data[malaria_data['Period'] == 2015].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [269]:
malaria_data[malaria_data['Period'] == 2014].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [270]:
malaria_data[malaria_data['Period'] == 2013].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [271]:
malaria_data[malaria_data['Period'] == 2012].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [272]:
malaria_data[malaria_data['Period'] == 2011].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [273]:
malaria_data[malaria_data['Period'] == 2010].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [274]:
malaria_data[malaria_data['Period'] == 2009].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [275]:
malaria_data[malaria_data['Period'] == 2008].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [276]:
malaria_data[malaria_data['Period'] == 2007].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [277]:
malaria_data[malaria_data['Period'] == 2006].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [278]:
malaria_data[malaria_data['Period'] == 2005].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [279]:
malaria_data[malaria_data['Period'] == 2004].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [280]:
malaria_data[malaria_data['Period'] == 2003].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [281]:
malaria_data[malaria_data['Period'] == 2002].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [282]:
malaria_data[malaria_data['Period'] == 2001].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64

In [283]:
malaria_data[malaria_data['Period'] == 2000].count()

Location         107
Indicator        107
Period           107
First Tooltip    107
dtype: int64